x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/manage_processor_form/processors/script.tsx

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { EuiCode, EuiSwitch, EuiFormRow } from '@elastic/eui';
+
+import { FIELD_TYPES, fieldValidators, UseField, Field } from '../../../../../../shared_imports';
+
+import { XJsonEditor, TextEditor } from '../field_components';
+
+import { FieldsConfig, to, from, FormFieldsComponent } from './shared';
+
+const { isJsonField, emptyField } = fieldValidators;
+
+const fieldsConfig: FieldsConfig = {
+  /* Required fields config */
+
+  id: {
+    type: FIELD_TYPES.TEXT,
+    deserializer: String,
+    label: i18n.translate(
+      'xpack.ingestPipelines.pipelineEditor.scriptForm.storedScriptIDFieldLabel',
+      {
+        defaultMessage: 'Stored script ID',
+      }
+    ),
+    helpText: i18n.translate(
+      'xpack.ingestPipelines.pipelineEditor.scriptForm.storedScriptIDFieldHelpText',
+      {
+        defaultMessage: 'Stored script reference.',
+      }
+    ),
+    validations: [
+      {
+        validator: emptyField(
+          i18n.translate('xpack.ingestPipelines.pipelineEditor.scriptForm.idRequiredError', {
+            defaultMessage: 'A value is required.',
+          })
+        ),
+      },
+    ],
+  },
+
+  source: {
+    type: FIELD_TYPES.TEXT,
+    deserializer: String,
```

Comment:
>  might lead to errors

Hm, I'd be curious to know how it might lead to errors - at least in the case of Painless I can't see how it would. I would prefer to not alter the user provided source code as far as possible.

Additionally, if users are using our debugging workflow (which they really should 😄) they should be able track down syntax related script issues.

I'm going to punt on this one for now - at least until we have a clear case for how this might lead to issues.

Reply:
> would prefer to not alter the user provided source code as far as possible

Good point! I agree to not do any editing of user input unless there is a clear issue for it.